x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/mapper/HllFieldMapper.java

```diff
@@ -0,0 +1,737 @@
+                            throw new MapperParsingException("error parsing field ["
+                                + name() + "], ["+ SKETCH_FIELD + "] elements must be >= 0 but got " + newValue);
+                        }
+                        if (newValue > Byte.MAX_VALUE) { // is that correct? what is the real max value for a runLen?
+                            throw new MapperParsingException("error parsing field ["
+                                + name() + "], ["+ SKETCH_FIELD + "] elements must be <= " + Byte.MAX_VALUE + " but got " + newValue);
+                        }
+                        runLens.add((byte) newValue);
+                        token = subParser.nextToken();
+                    }
+                } else {
+                    throw new MapperParsingException("error parsing field [" +
+                        name() + "], with unknown parameter [" + fieldName + "]");
+                }
+                token = subParser.nextToken();
+            }
+            if (runLens == null) {
+                throw new MapperParsingException("error parsing field ["
+                    + name() + "], expected field called [" + SKETCH_FIELD.getPreferredName() + "]");
+            }
+            if (runLens.size() != m) {
+                throw new MapperParsingException("error parsing field ["
+                    + name() + "], expected length from [" + SKETCH_FIELD.getPreferredName() +"] is " + m + ""
+                    + ", got [" + runLens.size() + "]");
+            }
+            if (fieldType().hasDocValues()) {
+                final ByteBuffersDataOutput dataOutput = new ByteBuffersDataOutput();
+                writeCompressed(runLens, dataOutput);
+                final BytesRef docValue = new BytesRef(dataOutput.toArrayCopy(), 0, Math.toIntExact(dataOutput.size()));
+                final Field field = new BinaryDocValuesField(name(), docValue);
+                if (context.doc().getByKey(fieldType().name()) != null) {
+                    throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() +
+                        "] doesn't not support indexing multiple values for the same field in the same document");
+                }
+                context.doc().addWithKey(fieldType().name(), field);
+            }
+
+        } catch (Exception ex) {
+            if (ignoreMalformed.value() == false) {
+                throw new MapperParsingException("failed to parse field [{}] of type [{}]",
+                    ex, fieldType().name(), fieldType().typeName());
+            }
+
+            if (subParser != null) {
+                // close the subParser so we advance to the end of the object
+                subParser.close();
+            }
+            context.addIgnoredField(fieldType().name());
+        }
+        context.path().remove();
+    }
+
+    private void writeCompressed(ByteArrayList runLens, ByteBuffersDataOutput dataOutput) throws IOException {
```

Comment:
Could we add a comment/javadoc somewhere (maybe here?  Maybe at top with the constants?) about how the different compression scheme work or differ?  Fixed and run-length make sense, but the "custom" confused me :)

Reply:
Ah I see, they are mentioned below in the class javadocs.  Ok so maybe this isn't necessary here :)